File: IOS开发指南 从零基础到App Store上架_Source/ch16/16.2.1/MyNotes（ARC工程使用MRC代码）/MyNotes/MasterViewController.m
//
//  MasterViewController.m
//  MyNotes
//
//  Created by 关东升 on 12-9-26.
//  本书网站：http://www.iosbook1.com
//  智捷iOS课堂：http://www.51work6.com
//  智捷iOS课堂在线课堂：http://v.51work6.com
//  智捷iOS课堂新浪微博：http://weibo.com/u/3215753973
//  作者微博：http://weibo.com/516inc
//  官方csdn博客：http://blog.csdn.net/tonny_guan
//  QQ：1575716557 邮箱：jylong06@163.com
//

#import "MasterViewController.h"
#import "DetailViewController.h"

@implementation MasterViewController


- (void)awakeFromNib
{
    if ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPad) {
        self.clearsSelectionOnViewWillAppear = NO;
        self.contentSizeForViewInPopover = CGSizeMake(320.0, 600.0);
    }
    [super awakeFromNib];
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.navigationItem.leftBarButtonItem = self.editButtonItem;
    
    self.detailViewController = (DetailViewController *)[[self.splitViewController.viewControllers lastObject] topViewController];
    
    [self startRequest];
}


- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


#pragma mark - Table View

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return self.listData.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"Cell"  forIndexPath:indexPath];
    
    NSMutableDictionary*  dict = self.listData[indexPath.row];
    cell.textLabel.text = [dict objectForKey:@"Content"];
    cell.detailTextLabel.text = [dict objectForKey:@"CDate"];
    
    return cell;
}

- (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath
{
    // Return NO if you do not want the specified item to be editable.
    return YES;
}

- (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath
{
    if (editingStyle == UITableViewCellEditingStyleDelete) {
        
        //        Note *note = [self.listData objectAtIndex:[indexPath row]];
        //        NoteBL *bl = [[NoteBL alloc] init];
        //        self.listData = [bl remove: note];
        
        //[tableView deleteRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationFade];
        
    } else if (editingStyle == UITableViewCellEditingStyleInsert) {
        
    }
}


- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    if ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPad) {
        //        Note  *note = self.listData[indexPath.row];
        //        self.detailViewController.detailItem = note;
    }
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    if ([[segue identifier] isEqualToString:@"showDetail"]) {
        //       NSIndexPath *indexPath = [self.tableView indexPathForSelectedRow];
        //        Note  *note = self.listData[indexPath.row];
        //        [[segue destinationViewController] setDetailItem:note];
    }
}

/*
 * 开始请求Web Service
 */
-(void)startRequest
{   
    NSString *strURL = [[NSString alloc] initWithFormat:@"http://iosbook1.com/service/mynotes/webservice.php?email=%@&type=%@&action=%@",@"test@51work6.com",@"JSON",@"query"];
    
	NSURL *url = [NSURL URLWithString:[strURL URLEncodedString]];
	
    ASIHTTPRequest *request = [ASIHTTPRequest requestWithURL:url];
    [request startSynchronous];
    NSLog(@"请求完成...");
    NSError *error = [request error];
    
    if (!error) {
        //NSString *response = [request responseString];
        NSData *data  = [request responseData];
        NSDictionary *resDict = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingAllowFragments error:nil];
        [self reloadView:resDict];
    }
}

//重新加载表视图
-(void)reloadView:(NSDictionary*)res
{
    NSNumber *resultCodeObj = [res objectForKey:@"ResultCode"];
    if ([resultCodeObj integerValue] >=0)
    {
        self.listData = [res objectForKey:@"Record"];
        [self.tableView reloadData];
    } else {
        NSString *errorStr = [resultCodeObj errorMessage];
        UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:@"错误信息"
                                                            message:errorStr
                                                           delegate:nil
                                                  cancelButtonTitle:@"OK"
                                                  otherButtonTitles: nil];
        [alertView show];
    }
    
    
}


@end

File: IOS8应用开发入门经典_Source/iOS8Projects/Hour 14/Projects/FlowerDetail/FlowerDetail/MasterViewController.swift
//
//  MasterViewController.swift
//  FlowerDetail
//
//  Created by John Ray on 12/3/14.
//  Copyright (c) 2014 John E. Ray. All rights reserved.
//

import UIKit

class MasterViewController: UITableViewController {

    var detailViewController: DetailViewController? = nil
    var objects = NSMutableArray()
    
    var flowerData: [AnyObject] = []
    var flowerSections: [String] = []



    override func awakeFromNib() {
        super.awakeFromNib()
        if UIDevice.currentDevice().userInterfaceIdiom == .Pad {
            self.clearsSelectionOnViewWillAppear = false
            self.preferredContentSize = CGSize(width: 320.0, height: 600.0)
        }
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        createFlowerData()
        
        /*self.navigationItem.leftBarButtonItem = self.editButtonItem()

        let addButton = UIBarButtonItem(barButtonSystemItem: .Add, target: self, action: "insertNewObject:")
        self.navigationItem.rightBarButtonItem = addButton
        */
        if let split = self.splitViewController {
            let controllers = split.viewControllers
            self.detailViewController = controllers[controllers.count-1].topViewController as? DetailViewController
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func insertNewObject(sender: AnyObject) {
        objects.insertObject(NSDate(), atIndex: 0)
        let indexPath = NSIndexPath(forRow: 0, inSection: 0)
        self.tableView.insertRowsAtIndexPaths([indexPath], withRowAnimation: .Automatic)
    }

    // MARK: - Segues

    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        if segue.identifier == "showDetail" {
            if let indexPath = self.tableView.indexPathForSelectedRow() {
                //let object = objects[indexPath.row] as NSDate
                let controller = (segue.destinationViewController as UINavigationController).topViewController as DetailViewController
                //controller.detailItem = object
                controller.detailItem=self.flowerData[indexPath.section][indexPath.row]
                controller.navigationItem.leftBarButtonItem = self.splitViewController?.displayModeButtonItem()
                controller.navigationItem.leftItemsSupplementBackButton = true
            }
        }
    }

    // MARK: - Table View

    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return flowerSections.count
    }
    
    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        //return objects.count
        return flowerData[section].count
    }

    override func tableView(tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
        return flowerSections[section]
    }
    
    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCellWithIdentifier("flowerCell", forIndexPath: indexPath) as UITableViewCell

        //let object = objects[indexPath.row] as NSDate
        //cell.textLabel!.text = object.description
        
        cell.textLabel!.text = flowerData[indexPath.section][indexPath.row]["name"] as String!
        cell.detailTextLabel!.text = flowerData[indexPath.section][indexPath.row]["url"] as String!
        
        cell.imageView!.image = UIImage(named: flowerData[indexPath.section][indexPath.row]["picture"] as String!)!
        
        return cell
    }

    override func tableView(tableView: UITableView, canEditRowAtIndexPath indexPath: NSIndexPath) -> Bool {
        // Return false if you do not want the specified item to be editable.
        return false
    }
    
    func createFlowerData() {
        var redFlowers: [Dictionary<String,String>] = []
        var blueFlowers: [Dictionary<String,String>] = []
        
        flowerSections = ["Red Flowers","Blue Flowers"]
        
        redFlowers.append(["name":"Poppy","picture":"Poppy.png","url":"http://en.wikipedia.org/wiki/Poppy"])
        redFlowers.append(["name":"Tulip","picture":"Tulip.png","url":"http://en.wikipedia.org/wiki/Tulip"])
        redFlowers.append(["name":"Gerbera","picture":"Gerbera.png","url":"http://en.wikipedia.org/wiki/Gerbera"])
        redFlowers.append(["name":"Peony","picture":"Peony.png","url":"http://en.wikipedia.org/wiki/Peony"])
        redFlowers.append(["name":"Rose","picture":"Rose.png","url":"http://en.wikipedia.org/wiki/Rose"])
        redFlowers.append(["name":"Hollyhock","picture":"Hollyhock.png","url":"http://en.wikipedia.org/wiki/Hollyhock"])
        redFlowers.append(["name":"Straw Flower","picture":"Strawflower.png","url":"http://en.wikipedia.org/wiki/Peony"])
        
        blueFlowers.append(["name":"Hyacinth","picture":"Hyacinth.png","url":"http://en.wikipedia.org/wiki/Hyacinth_(flower)"])
        blueFlowers.append(["name":"Hydrangea","picture":"Hydrangea.png","url":"http://en.wikipedia.org/wiki/Hydrangea"])
        blueFlowers.append(["name":"Sea Holly","picture":"Sea Holly.png","url":"http://en.wikipedia.org/wiki/Sea_holly"])
        blueFlowers.append(["name":"Grape Hyacinth","picture":"Grape Hyacinth.png","url":"http://en.wikipedia.org/wiki/Grape_hyacinth"])
        blueFlowers.append(["name":"Phlox","picture":"Phlox.png","url":"http://en.wikipedia.org/wiki/Phlox"])
        blueFlowers.append(["name":"Pin Cushion Flower","picture":"Pincushion flower.png","url":"http://en.wikipedia.org/wiki/Scabious"])
        blueFlowers.append(["name":"Iris","picture":"Iris.png","url":"http://en.wikipedia.org/wiki/Iris_(plant)"])
        
        flowerData=[redFlowers,blueFlowers]
    }

    /*
    override func tableView(tableView: UITableView, commitEditingStyle editingStyle: UITableViewCellEditingStyle, forRowAtIndexPath indexPath: NSIndexPath) {
        if editingStyle == .Delete {
            objects.removeObjectAtIndex(indexPath.row)
            tableView.deleteRowsAtIndexPaths([indexPath], withRowAnimation: .Fade)
        } else if editingStyle == .Insert {
            // Create a new instance of the appropriate class, insert it into the array, and add a new row to the table view.
        }
    }
*/


}

